xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/resources/xwiki-image/plugin.js

```diff
@@ -229,14 +238,24 @@
          * @param key the property key to removew
          */
         function removeAttribute(widget, key) {
+          widget.element.removeAttribute(key);
           widget.parts.image.removeAttribute(key);
           widget.wrapper.removeAttribute(key);
         }
 
+        function updateFigureDimensions(widget) {
+          if (widget.data.hasCaption) {
+            setAttribute(widget, 'style', "width: " + widget.data.width + "px");
```

Comment:
Won't this override any existing styles on the figure?

Reply:
Good catch, thanks. I'll apply the same merge strategy as in the listeners.